hardware/battery/battery_management_system_protocol.ksy

```diff
@@ -0,0 +1,223 @@
+  endian: be
+
+doc: |
+  Many modern general purpose BMS include a UART/Bluetooth based communication interface.
+  After sending read requests they respond with various information's about the battery state in
+  a custom binary format.
+
+doc-ref: https://www.lithiumbatterypcb.com/Protocol%20English%20Version.rar
+
+seq:
+  - id: magic_start
+    contents: [0xdd]
+  - id: cmd
+    type: u1
+
+  - id: body
+    type:
+      switch-on: cmd
+      cases:
+        0xa5: read_req
+        0x5a: write_req
+        _: response(cmd)
+  
+  - id: checksum
+    size: 2
+  - id: magic_end
+    contents: [0x77]
+
+types:
+  read_req:
+    seq:
+      - id: req_cmd
+        type: u1
+        doc: Same value as cmd for response
+      - id: data_len
+        contents: [0x00]
+  write_req:
+    seq:
+      - id: req_cmd
+        type: u1
+        doc: Same value as cmd for response
+      - id: data_len
+        type: u1
+      - id: write_data
+        size: data_len
+
+  basic_info:
+    seq:
+      - id: total
+        type: voltage
+      - id: current
+        type: current
+      - id: remain_cap
+        type: capacity
+      - id: typ_cap
+        type: capacity
+      - id: cycles
+        type: u2
+        doc: Cycle times
+      - id: prod_date
+        type: u2
+        doc: Production date
+      - id: balance_status
+        type: balance_list
+        doc: List of balance bits
+      - id: prot_status
+        type: prot_list
+        size: 2
+        doc: List of protection bits
+      - id: software_version
+        type: u1
+      - id: remain_cap_percent
+        type: u1
+        doc: Portion of remaining capacity
+      - id: fet_status
+        type: fet_bits
+        size: 1
+      - id: cell_count
+        type: u1
+      - id: ntc_count
+        type: u1
+      - id: temps
+        type: temp
+        size: 2
+        repeat: expr
+        repeat-expr: ntc_count
+    enums:
+      fet_bit:
+        0: off
+        1: on
+    types:
+      balance_list:
+        seq:
+          - id: flag
+            type: b1
+            repeat: expr
+            repeat-expr: 32
+      prot_list:
+        seq:
+          - id: reserved
+            type: b3
+          - id: fet_lock
+            type: b1
+          - id: ic_error
+            type: b1
+          - id: ocp_short
+            type: b1
+          - id: ocp_discharge
+            type: b1
+          - id: ocp_charge
+            type: b1
+          - id: utp_discharge
+            type: b1
+          - id: otp_discharge
+            type: b1
+          - id: utp_charge
+            type: b1
+          - id: otp_charge
+            type: b1
+          - id: uvp_pack
+            type: b1
+          - id: ovp_pack
+            type: b1
+          - id: uvp_cell
+            type: b1
+          - id: ovp_cell
+            type: b1
+      fet_bits:
+        seq:
+          - id: reserved
+            type: b6
+          - id: discharge
+            type: b1
+            enum: fet_bit
```

Comment:
Isn't it already `b1`?

Reply:
Don't know, what is a enum per default? In any case, does it hurt to specify the type?

Reply:
> Don't know, what is a enum per default? In any case, does it hurt to specify the type?

I think that @KOLANICH meant that when you do just this

```diff
           - id: discharge
             type: b1
-            enum: fet_bit
```

, the attribute `discharge` will be a boolean (`true`/`false`), which is already enough to specify the value logic. Binding it with the enum `fet_bit`:

https://github.com/kaitai-io/kaitai_struct_formats/blob/042fc917646aa60d35edcd0ba100d4b05bbdf7c1/hardware/battery/battery_management_system_protocol.ksy#L97-L100

is redundant, because it just describes a non-standard duplicate of the `false`/`true` logic (`off` meaning `false` and `on` = `true`).

Reply:
Ahh now I understand, that is indeed redundant, removed in a623ed0.